src/modules/crud.js

```diff
@@ -0,0 +1,143 @@
+    return JSON.parse(possibleList);
+  }
+
+  return listArray;
+};
+
+const updateLocal = (newData) => {
+  localStorage.setItem('New Todo', JSON.stringify(newData));
+};
+
+export const loadTodo = () => {
+  list.innerHTML = '';
+  const local = getLocal();
+
+  local.forEach((element, i) => {
+    const li = document.createElement('li');
+
+    const checkbox = document.createElement('input');
+    checkbox.classList.add('checkbox');
+    checkbox.type = 'checkbox';
+
+    const paragraph = document.createElement('p');
+    paragraph.id = i;
+    paragraph.classList.add('paragraph');
+    paragraph.textContent = element.description;
+
+    const div = document.createElement('div');
+    div.classList.add('TextItems');
+
+    const remove = document.createElement('span');
+    remove.classList.add('remove');
+    // eslint-disable-next-line quotes
+    remove.innerHTML = `<i class="fas fa-trash delete-icon"></i>`;
+
+    li.appendChild(checkbox);
+    li.appendChild(paragraph);
+    li.appendChild(remove);
+    list.appendChild(li);
+
+    addInput.value = '';
+  });
+};
+
+export const createTodo = () => {
+  const text = addInput.value;
+
+  if (text === '') {
+    return;
+  }
+
+  listArray = getLocal();
+
+  listArray.push({
+    completed: false,
+    index: listArray.length + 1,
+    description: text,
+  });
+
+  updateLocal(listArray);
+  loadTodo();
+};
+
+export const updateTodo = () => {
+  const editInput = document.getElementsByName('editInput')[0];
+
+  if (!editInput) {
+    // eslint-disable-next-line no-useless-return
+    return;
+  }
+
+  const newText = editInput.value;
+
+  if (newText !== '') {
+    const paragraph = editInput.parentElement.querySelector('.paragraph');
+    const position = parseInt(paragraph.id, 10);
+
+    const local = getLocal();
+
+    const newList = [];
+
+    local.forEach((element, i) => {
+      if (position === i) {
+        newList.push({
+          index: element.index,
+          description: newText,
+          completed: element.completed,
+        });
+      } else {
+        newList.push(element);
+      }
+    });
+
+    updateLocal(newList);
+    loadTodo();
+  }
+
+  editInput.remove();
+};
+
+export const showEditInput = (paregraphElement) => {
+  const editInput = document.getElementsByName('editInput')[0];
+
+  if (editInput) {
+    editInput.remove();
+  }
+
+  const input = document.createElement('input');
+  input.type = 'text';
+  input.name = 'editInput';
+  input.value = paregraphElement.textContent;
+  input.classList.add('editInput');
+
+  paregraphElement.parentElement.appendChild(input);
+  input.focus();
+};
+
+export const removeTodo = (removeElement, index) => {
+  const getLocalStorageData = localStorage.getItem('New Todo');
+  const listArray = JSON.parse(getLocalStorageData);
+  listArray.splice(index, 1);
+  localStorage.setItem('New Todo', JSON.stringify(listArray));
+  removeElement.parentElement.parentElement.remove();
+};
```

Comment:
- [ ] I created three tasks and they all have the correct index [1,2,3]. However, when I deleted one task, the indexes of the remaining tasks did not update, that is the indexes were [2, 3] instead of [1,2].  Please let's make sure that deleting a task updates all remaining items' indexes, so they represent the current list order and are unique. 

- [ ] When I delete a task, the task gets deleted from the page but surprisingly, another task gets deleted from the local storage instead of the task I deleted. Let's fix this.

- [ ] Also when I added a new task, there appears to be two tasks with the same index. Please let's fix that. Please take a look below:


Before I deleted eat task:
<img width="913" alt="Loom_ZtviJfkqfY" src="https://user-images.githubusercontent.com/81468675/149551756-4a7c7c93-2b86-43d5-a126-198af700297c.png">

After I deleted eat task, I can still see eat task in the local storage:

<img width="875" alt="Loom_e2x1kSENoG" src="https://user-images.githubusercontent.com/81468675/149551833-f97bba19-e776-4bef-a8be-91cace638606.png">

When I added a new task:
<img width="871" alt="Loom_g4cGv7BDWA" src="https://user-images.githubusercontent.com/81468675/149552056-1e0a43fd-8b18-44a7-8581-8d9d22ae6417.png">